lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskStatus.cs

```diff
@@ -64,7 +66,7 @@ internal sealed class TaskStatus
         public TaskStatus(IHeartBeatManager heartBeatManager, string contextId, string taskId, Optional<ISet<ITaskMessageSource>> evaluatorMessageSources)
         {
             _heartBeatManager = heartBeatManager;
-            _taskLifeCycle = new TaskLifeCycle();
+            _taskLifeCycle = TangFactory.GetTang().NewInjector().GetInstance<TaskLifeCycle>();
```

Comment:
This is simply for the deprecation path, which would not have their task handlers invoked anyhow.